### PR TITLE
Switch to using coresmd instead of dnsmasq in podman-quadlets recipe

### PR DESCRIPTION
**Pull request by travisbcotton**
Adding coresmd to provide dhcp. Also removing dnsmasq data. 

coredhcp does not include a tftp server right now, still need to figure out if that is coming from coresmd or an external service.

### CODE REVIEW

File: lanl/podman-quadlets/inventory/group_vars/ochami/coredhcp.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+coredhcp_server_id: '{{ cluster_boot_ip }}'
+coredhcp_dns_server: '{{ cluster_boot_ip }}'
+coredhcp_router: '{{ cluster_boot_ip }}'
+coredhcp_netmask: '255.255.255.0'
+coredhcp_dhcp_pool: '172.16.0.200 172.16.0.250'
+coredhcp_tmp_lease_duration: '5m'
+coredhcp_lease_duration: '1h'
+coredhcp_cache_validity: '30s'
+coredhcp_ipxe_binary: 'ipxe-x86_64.efi'
```
Comment (synackd) [inline lanl/podman-quadlets/inventory/group_vars/ochami/coredhcp.yaml:9]: Is this needed? The newest CoreDHCP with coresmd will determine this based on the architecture and these binaries are now included in the container as ov v0.1.0.

File: lanl/podman-quadlets/inventory/group_vars/ochami/packages.yaml [hunk 2]
```diff
@@ -8,3 +8,4 @@ ochami_package_dependencies:
   - python3-requests
   - python3-yaml
   - python3-jwt
+  - ipxe-bootimgs-x86
```
Comment (synackd) [inline lanl/podman-quadlets/inventory/group_vars/ochami/packages.yaml:11]: Do we need this package anymore? The bootloaders should be included in the CoreDHCP container as of coresmd v0.1.0.

File: lanl/podman-quadlets/inventory/group_vars/ochami/quadlets.yaml [hunk 3]
```diff
@@ -324,38 +322,19 @@ _haproxy_quadlets:
       - cloud-init-server.service
       - tpm-manager.service
 
-_dnsmasq_quadlets:
-  - name: dnsmasq
-    image: ghcr.io/openchami/dnsmasq:v0.1.11
+_coresmd_quadlets:
+  - name: coresmd
+    image: ghcr.io/openchami/coredhcp:v0.1.1
```
Comment (synackd) [inline lanl/podman-quadlets/inventory/group_vars/ochami/quadlets.yaml:327]: If you want to not have an external reboot.ipxe (and thus have to have an external TFTP server), you can upgrade to v0.2.0. See my comment in the configs/templates/coredhcp/coredhcp.yaml.j2 file.

File: lanl/podman-quadlets/roles/configs/templates/coredhcp/coredhcp.yaml.j2 [hunk 4]
```diff
@@ -0,0 +1,8 @@
+server4:
+  plugins:
+    - server_id: {{ coredhcp_server_id }}
+    - dns: {{ coredhcp_dns_server }}
+    - router: {{ coredhcp_router }}
+    - netmask: {{ coredhcp_netmask }}
+    - coresmd: https://{{ cluster_name }}.{{ cluster_domain }}:8443 http://{{ cluster_boot_ip }}:8081 /root_ca/root_ca.crt {{ coredhcp_cache_validity }} {{ coredhcp_lease_duration }}
+    - bootloop: /tmp/coredhcp.db {{ coredhcp_tmp_lease_duration }} {{ coredhcp_dhcp_pool }}
```
Comment (synackd) [inline lanl/podman-quadlets/roles/configs/templates/coredhcp/coredhcp.yaml.j2:8]: If you want to use CoreDHCP with coresmd v0.2.0, the only change you'll need to make is on this line. You only need to add `default` as the second argument to tell the built-in TFTP server to serve the built-in reboot iPXE script.

Replace:
```yaml
- bootloop: /tmp/coredhcp.db {{ coredhcp_tmp_lease_duration }} {{ coredhcp_dhcp_pool }}
```
with:
```yaml
- bootloop: /tmp/coredhcp.db default {{ coredhcp_tmp_lease_duration }} {{ coredhcp_dhcp_pool }}
```

**Review by synackd — COMMENTED** [review]
Finished with my initial review. I don't think it's too big of a jump to use the newest (v0.2.0) coresmd as it only requires one small config change, and we can get rid of the iPXE bootloader package/files.

Let me know what you think.

File: lanl/podman-quadlets/inventory/group_vars/ochami/coredhcp.yaml (hunk 1 above)
Comment (travisbcotton) [reply lanl/podman-quadlets/inventory/group_vars/ochami/coredhcp.yaml]: Should be removed now.

File: lanl/podman-quadlets/roles/configs/templates/coredhcp/coredhcp.yaml.j2 (hunk 4 above)
Comment (travisbcotton) [reply lanl/podman-quadlets/roles/configs/templates/coredhcp/coredhcp.yaml.j2]: Added a variable `coredhcp_custom_ipxe` that is set to default. Updated the coredhcp template to use it

File: lanl/podman-quadlets/inventory/group_vars/ochami/quadlets.yaml (hunk 3 above)
Comment (travisbcotton) [reply lanl/podman-quadlets/inventory/group_vars/ochami/quadlets.yaml]: Bumped the version to v0.2.0

**Pull request merged**